webapp/api/advantage.py

```diff
@@ -0,0 +1,102 @@
+import os
+
+import flask
+
+from requests import Request, Session
+from pymacaroons import Macaroon
+
+from webapp.macaroons import binary_serialize_macaroons
+
+
+API_URL = os.getenv("ADVANTAGE_API", "https://contracts.canonical.com/")
+api_session = Session()
+
+
+class UnauthorizedRequest(Exception):
+    pass
+
+
+def _prepare_request(method, path, data=None, session=None):
+    request = Request(method=method, url=f"{API_URL}{path}")
+
+    if session:
+        root = Macaroon.deserialize(flask.session["macaroon_root"])
+        discharge = Macaroon.deserialize(flask.session["macaroon_discharge"])
+        bound = root.prepare_for_request(discharge)
+        token = binary_serialize_macaroons([root, bound]).decode("utf-8")
+        request.headers["Authorization"] = f"Macaroon {token}"
+
+    if data is not None:
```

Comment:
Can you not do `if data:` here? Why check for None here and not on line 22 above?

Reply:
No because we need to call an endpoint with data={} and `{}` would evaluate to `False`.